src/utils/client.js

```diff
@@ -273,14 +278,19 @@ export default class Client {
     });
   }
 
-  submitCustom(customKeyValues) {
+  submitCustom(customKeyValues, signatureRef) {
     const { userKey } = initInfo();
     const url = `${API_URL}panel/custom/submit/v2/`;
 
     const jsonKeyVals = {};
     const formData = new FormData();
     Object.entries(customKeyValues).forEach(([key, val]) => {
-      if (val instanceof Blob) {
+      if (isBase64PNG(val)) {
+        if (signatureRef && signatureRef[key])
```

Comment:
i don't think any other fromDataURL logic is necessary other than the one in the hydration function to keep it clean. we can establish a constraint that the value has to be set before the step with the signature field is loaded, toherwise it won't update in real time

Reply:
this logic is (at least currently) needed to set the canvas & file when the user submits a base64 via `setValues`